sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -425,11 +425,15 @@ object IntervalUtils {
   }
 
   private object ParseState extends Enumeration {
+    type ParseState = Value
+
     val PREFIX,
         BEGIN_VALUE,
```

Comment:
we can rename it to `TRIM_BEFORE_PARSE_VALUE`

Reply:
so we may change `PARSE_UNIT_VALUE` to `PARSE_VALUE` too

How about renaming them all as,
```scala
    val PREFIX,
        BODY,
        SIGN,
        TRIM_BEFORE_VALUE,
        VALUE,
        VALUE_FRACTIONAL_PART,
        TRIM_BEFORE_UNIT,
        BEGIN_UNIT,
        UNIT_SUFFIX,
        END_UNIT = Value
```
Seems loud and clear enough for each state

Reply:
what does `BODY` mean? others LGTM

Reply:
interval (PREFIX,  BODY)
BODY (SIGN,  VALUE , UNIT)+

Seems not  persuadable enough

Reply:
or `NEXT_VALUE_UNIT`?